src/test/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjectorForIndexingTest.java

```diff
@@ -0,0 +1,203 @@
+                          }
+                      """;
+
+      // Try with overlapping paths
+      assertAllowProjection(Arrays.asList("a", "a.b"), INPUT_DOC, EXP_OUTPUT);
+      assertAllowProjection(Arrays.asList("a.b", "a"), INPUT_DOC, EXP_OUTPUT);
+      assertAllowProjection(Arrays.asList("a.b.z", "a.b"), INPUT_DOC, EXP_OUTPUT);
+      assertAllowProjection(Arrays.asList("a.b", "a.b.z"), INPUT_DOC, EXP_OUTPUT);
+      assertAllowProjection(Arrays.asList("a.b.z", "a"), INPUT_DOC, EXP_OUTPUT);
+    }
+  }
+
+  @Nested
+  class DenyFiltering {
+    @Test
+    public void testRootDeny() {
+      assertDenyProjection(
+          Arrays.asList("d", "b", "x"), // "x" won't match anything
+          """
+                        { "_id":123, "a": 5, "b": 6, "c": 7, "d": 8 }
+                    """,
+          """
+                        { "_id":123, "a": 5, "c": 7 }
+                    """);
+    }
+
+    @Test
+    public void testNestedDeny() {
+      assertDenyProjection(
+          Arrays.asList("a.y", "c", "z"), // "z" non-matching
+          """
+                      { "_id":123,
+                        "a": { "x":1, "y":2 },
+                        "b": { "x":3, "y":4 },
+                        "c": { "x":2, "y":3 }
+                      }
+                  """,
+          """
+                        { "_id":123,
+                          "a": { "x":1 },
+                          "b": { "x":3, "y":4 }
+                        }
+                    """);
+    }
+
+    @Test
+    public void testOverlappingDeny() {
+      final String INPUT_DOC =
+          """
+                          {
+                            "_id":123,
+                            "a": {
+                               "b" : {
+                                  "z" : true,
+                                  "abc": 123
+                                }
+                             },
+                             "b" : 123
+                          }
+                      """;
+      final String EXP_OUTPUT =
+          """
+                          {
+                            "_id":123,
+                             "b" : 123
+                          }
+                      """;
+
+      // Try with overlapping paths
+      assertDenyProjection(Arrays.asList("a", "a.b"), INPUT_DOC, EXP_OUTPUT);
+      assertDenyProjection(Arrays.asList("a.b", "a"), INPUT_DOC, EXP_OUTPUT);
+      assertDenyProjection(Arrays.asList("a.b.z", "a"), INPUT_DOC, EXP_OUTPUT);
+      assertDenyProjection(Arrays.asList("a", "a.b.z"), INPUT_DOC, EXP_OUTPUT);
+    }
+  }
+
+  @Nested
+  class SpecialCases {
+    @Test
+    public void testAllowAll() {
+      final String DOC =
+          """
+              { "a": 5, "b": { "enabled": true}, "c": 7, "d": [ { "value": 42} ] }
+          """;
+      // First with empty Sets:
+      assertAllowProjection(Arrays.asList(), DOC, DOC);
+      // And then "*" notation
+      assertAllowProjection(Arrays.asList("*"), DOC, DOC);
```

Comment:
Not a valid case.

Reply:
Related to earlier comment: so, correct, depends on our definition.